packages/@lwc/engine/src/framework/restrictions.ts

```diff
@@ -432,6 +439,29 @@ function getLightningElementPrototypeRestrictionsDescriptors(
                 return originalDispatchEvent.apply(this, arguments);
             },
         }),
+        isConnected: generateAccessorDescriptor({
+            get(this: LightningElement) {
+                const vm = getAssociatedVM(this);
+                const componentTag = getComponentTag(vm);
+                assert.isFalse(
+                    isBeingConstructed(vm),
+                    `this.isConnected cannot be accessed during the construction phase of the custom` +
+                        ` element for ${componentTag} because it is redundant. The value will always be` +
```

Comment:
Even if the error makes sense in 99% of the case, I would like to bring more nuance to this statement. If the element is already connected to the DOM before the upgrade the `isConnected` will be true in the `constructor`: https://jsfiddle.net/pmdartus/fb1Lt8wa/2/